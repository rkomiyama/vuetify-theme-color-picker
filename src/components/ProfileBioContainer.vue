<template>
  <v-content>
    <v-layout fill-height>
      <v-card class="ma-auto">
        <v-img :src="user.avatar_url"></v-img>
        <v-card-title>{{ user.name }}</v-card-title>
        <v-card-text>
          <span>Username: {{ user.login }}</span><br>
          <a :href="user.html_url">{{ user.html_url }}</a><br>
          <span>Public Repos: {{ user.public_repos }}</span><br>
          <span>Created at {{ user.created_at | formatDate }}</span>
        </v-card-text>
      </v-card>
    </v-layout>
  </v-content>
</template>

<script>
import moment from 'moment';

export default {
  name: 'ProfileBioContainer',
  props: {
    user: {
      name: String,
      login: String,
      avatar_url: String,
      html_url: String,
      public_repos: Number,
      created_at: String
    }
  },
  filters: {
    formatDate (date) {
      return moment(date).format("LLL")
    }
  }
}
</script>
