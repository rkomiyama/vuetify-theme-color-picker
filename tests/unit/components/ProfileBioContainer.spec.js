import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer';
import faker from 'faker/locale/en_US'
import ProfileBioContainer from '@/components/ProfileBioContainer.vue'

Vue.use(Vuetify)
let wrapper
faker.seed(42)

describe('ProfileBioContainer.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(
      ProfileBioContainer, {
        propsData: {
          user: {
            name: faker.name.findName(),
            login: faker.internet.userName(),
            avatar_url: faker.internet.url(),
            html_url: faker.internet.url(),
            public_repos: faker.random.number(),
            created_at: "2010-07-28T22:19:56Z"
          }
        }
      }
    );
  });

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  it('ProfileBioContainer renders the same HTML', () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
    if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
})
