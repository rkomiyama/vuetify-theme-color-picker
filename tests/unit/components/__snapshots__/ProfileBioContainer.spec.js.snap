// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBioContainer.vue ProfileBioContainer renders the same HTML 1`] = `
<v-content-stub tag="main">
  <v-layout-stub tag="div" fill-height="">
    <v-card-stub loaderHeight="4" tag="div" class="ma-auto">
      <v-img-stub position="center center" src="https://riley.org" transition="fade-transition"></v-img-stub>
      <v-card-title-stub>Bryon Quigley</v-card-title-stub>
      <v-card-text-stub><span>Username: Mac96</span><br> <a href="http://abagail.us">http://abagail.us</a><br> <span>Public Repos: 99221</span><br> <span>Created at July 28, 2010 6:19 PM</span></v-card-text-stub>
    </v-card-stub>
  </v-layout-stub>
</v-content-stub>
`;
